graylog2-server/src/main/java/org/graylog2/rest/resources/system/DeflectorResource.java

```diff
@@ -97,16 +99,19 @@ public DeflectorSummary deflector(@ApiParam(name = "indexSetId") @PathParam("ind
     @AuditEvent(type = AuditEventTypes.ES_WRITE_INDEX_UPDATE_JOB_START)
     @Deprecated
     public void deprecatedCycle() {
-        final IndexSet indexSet = indexSetRegistry.getDefault()
-                .orElseThrow(() -> new NotFoundException("Default index set not found"));
+        try {
+            final IndexSet indexSet = indexSetRegistry.getDefault();
 
-        checkCycle(indexSet);
+            checkCycle(indexSet);
 
-        final String msg = "Cycling deflector for default index set <" + indexSet.getConfig().id()  + ">. Reason: REST request.";
-        LOG.info(msg);
-        activityWriter.write(new Activity(msg, DeflectorResource.class));
+            final String msg = "Cycling deflector for default index set <" + indexSet.getConfig().id() + ">. Reason: REST request.";
+            LOG.info(msg);
+            activityWriter.write(new Activity(msg, DeflectorResource.class));
 
-        indexSet.cycle();
+            indexSet.cycle();
+        } catch (IllegalStateException e) {
+            throw new NotFoundException("Default index set not found");
```

Comment:
I would throw an `InternalServerErrorException` here (or simply let the `IllegalStateException` bubble up which results in the same) because it's actually an error in the Graylog system, not a wrong request by the client.